### PR TITLE
debug_out() cleanups

### PR DESCRIPTION
**Pull request by dirkmueller**
Passing 3 parameters is code size wise an overhead over two parameters.
We can easily refactor this into:

  debug_out(..,.., 1) -> debug_outln(.., ..);
  debug_out(..,.., 0) -> debug_print(.., ..);

which are functions with the same length (so vertical aligning is the
same). this reduces size of the result by ~ 800 bytes.

### CODE REVIEW

**Comment by ricki-z** [comment]
include paths without a leading "./" may result in usage of the wrong includes if there is a file with the same name in the path. Some it's more secure to use "./"
Naming functions: Why not naming the debug_out() replacements debug_out() and debug_outln()?
The difference between the two functions is the line break, the function names should show this (like write() and writeln()).
And a question: Do you know why the functions need less memory than one function with one more parameter?

File: airrohr-firmware/airrohr-firmware.ino [hunk 1]
```diff
@@ -604,22 +606,31 @@ const char data_first_part[] PROGMEM = "{\"software_version\": \"{v}\", \"sensor
 /*****************************************************************
  * Debug output                                                  *
  *****************************************************************/
-void debug_out(const String& text, const int level, const bool linebreak) {
+
```
Comment (ricki-z) [inline airrohr-firmware/airrohr-firmware.ino]: "if 0": Really?

**Comment by dirkmueller** [comment]
@ricki-z I am not sure I understand the comment about #include "./foo" vs #include "foo". both do exactly the same: look in local dir (relative to the file with the include) first, and only then traverse path. see https://gcc.gnu.org/onlinedocs/cpp/Search-Path.html and https://stackoverflow.com/questions/21593/what-is-the-difference-between-include-filename-and-include-filename

**Comment by dirkmueller** [comment]
regarding function naming: I had it originally debug_out and debug_outln, as I also liked that more. the issue is that then the arguments are no longer vertically aligned. I'm okay with that personally. 

the reason this is smaller is because we don't have to push a 3rd parameter into a register/onto the stack, which requires less machine code. we simply call two different functions, and as we get rid of the if() condition then it should be marginally faster (no practical difference though) as well.

File: airrohr-firmware/airrohr-firmware.ino (hunk 1 above)
Comment (dirkmueller) [reply airrohr-firmware/airrohr-firmware.ino]: leftover. I only used this to measure the potential to spend some time on optimizing debug() statements. when removing all of them over 25kb of progmem is removed, so there is a very large potential. i've removed this now.

**Comment by ricki-z** [comment]
#include "./foo" vs #include "foo"
Yes, the compiler SHOULD look first in the local directory. But the Arduino IDE seems to ignore this sometimes. We had this with the oledfonts.h and the DHT.h. In both cases the library versions were used if we haven't included "./foo.h"
So just to be sure I would like to this version.

**Comment by dirkmueller** [comment]
@ricki-z I removed the include ./foo cleanup commit from the patch series. thanks for the patience.

**Comment by ricki-z** [comment]
For clarification: I would like to change the includes. We used the shorter version as we switched to a 'local' copy of the DHT library. But it wasn't working with that, only with the './foo' version. And nobody knows what Arduino may change in future versions ;-)

**Pull request merged**